exec/cnex/lib/file_posix.c

```diff
@@ -0,0 +1,383 @@
+#ifdef __APPLE__
+#include <copyfile.h>
+#else
+// The following is needed for fchmod() on Posix
+#define  _POSIX_C_SOURCE 200809L
+#endif
+#include <dirent.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <utime.h>
+
+#include "enums.h"
+#include "exec.h"
+#include "cell.h"
+#include "nstring.h"
+#include "stack.h"
+
+
+
+static void handle_error(TExecutor *exec, int error, const char *path)
+{
+    switch (error) {
+        case EACCES: exec->rtl_raise(exec, "FileException.PermissionDenied", path, BID_ZERO);      break;
+        case EEXIST: exec->rtl_raise(exec, "FileException.DirectoryExists", path, BID_ZERO);       break;
+        case ENOENT: exec->rtl_raise(exec, "FileException.PathNotFound", path, BID_ZERO);          break;
+        default: {
+            char err[260];
+            sprintf(err, "%s: %s", path, strerror(error));
```

Comment:
And then using an unbounded `sprintf` with arbitrary input. Stack smashing!

Reply:
Fixed.